src/System.Private.CoreLib/shared/System/ByReference.cs

```diff
@@ -11,11 +11,19 @@ namespace System
     // around lack of first class support for byref fields in C# and IL. The JIT and 
     // type loader has special handling for it that turns it into a thin wrapper around ref T.
     [NonVersionable]
-    internal readonly ref struct ByReference<T>
+    internal
+#if !PROJECTN
+    readonly
+#endif
+    ref struct ByReference<T>
     {
         // CS0169: The private field '{blah}' is never used
 #pragma warning disable 169
-         private readonly IntPtr _value;
+         private
+#if !PROJECTN // readonly breaks codegen contract and asserts UTC
```

Comment:
Is this part really needed? 

I would expect that the problem is caused by just the `readonly` marker on the struct. I would not expect the `readonly` modifier on this field to make a difference.

Reply:
I would have to re-test this, but I thought `readonly struct` is just about the C# compiler placing a custom attribute on the struct that nobody cares about. Am I missing something?

Reply:
The IL generated against readonly and non-readonly structs is different. `readonly` is quite a bit more than just a custom attribute that nobody cares about. It chances performance characteristics of the generated IL.

Reply:
Seems to work okay with the `initonly` field.